From 4dff6b1119930de2b7037eb2ec495763f1834b06 Mon Sep 17 00:00:00 2001
From: Bret Sepulveda (bsep) <bsep@chromium.org>
Date: Wed, 05 Jun 2024 13:13:08 +0000
Subject: [PATCH] Revert changing DCHECK to CHECK in RenderWidgetHostViewAura::OnDeviceScaleFactorChanged.

While we should probably handle the violated assumption properly, we
also should not crash in the meantime.

Bug: crbug.com/337612968
Change-Id: I965893b8a3b9089be9936f184078c293c4dd74d1
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5600753
Reviewed-by: Jonathan Ross <jonross@chromium.org>
Commit-Queue: Bret Sepulveda <bsep@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1310570}
---

diff --git a/content/browser/renderer_host/render_widget_host_view_aura.cc b/content/browser/renderer_host/render_widget_host_view_aura.cc
index 1d00ea3..68ac126 100644
--- a/content/browser/renderer_host/render_widget_host_view_aura.cc
+++ b/content/browser/renderer_host/render_widget_host_view_aura.cc
@@ -2035,7 +2035,9 @@
   // Sometimes GetDisplayNearestWindow returns the default monitor. We don't
   // want to use that here.
   if (display.is_valid()) {
-    CHECK_EQ(new_device_scale_factor, display.device_scale_factor());
+    // TODO: crbug.com/337612968 - This assumption is not valid, and is probably
+    // causing bugs in other places.
+    DCHECK_EQ(new_device_scale_factor, display.device_scale_factor());
     current_cursor_.SetDisplayInfo(display);
   }
 }
